### PR TITLE
Bulk transfers to NodeDelegator and EigenLayer

### PR DESCRIPTION
**Pull request by naddison36**
Contract changes:
* `LRTDepositPool` added `transferAssetsToNodeDelegator` that is only Operator for many assets
* `LRTDepositPool` changed `transferAssetToNodeDelegator` to only Operator for single asset
* `NodeDelegator` added `depositAssetsIntoStrategy` that is only Operator for many assets
* `NodeDelegator` changed `depositAssetIntoStrategy` to only Operator for single asset

* Updated Unit Tests for the above changes
* Added Fork Tests for new multi asset functions
* Added `ContractUpgrades` helper contract for testing `LRTDepositPool` and `NodeDelegator` upgrades

The two new contracts can be deployed with the `UpgradeDepositPoolNodeDelegator` forge script. This has the following make commands

```
make upgrade-deposit-delegator-mainnet
make upgrade-deposit-delegator-fork
make upgrade-deposit-delegator-local
```

The 3/7 multisig needs to call the `ProxyAdmin` contract to upgrade each proxy to its new implementation.
* ProxyAdmin 0xF83cacA1bC89e4C7f93bd17c193cD98fEcc6d758
* LRTDepositPool proxy 0xA479582c8b64533102F6F528774C536e354B8d32
* NodeDelegator proxy 0x8bBBCB5F4D31a6db3201D40F478f30Dc4F704aE2

### CODE REVIEW

File: contracts/LRTDepositPool.sol [hunk 1]
```diff
@@ -319,6 +319,44 @@ contract LRTDepositPool is ILRTDepositPool, LRTConfigRoleChecker, PausableUpgrad
         }
     }
 
+    /// @notice Transfers all specified assets lying in this DepositPool to a node delegator
+    /// @dev only callable by LRT Manager
```
Comment (shahthepro) [inline contracts/LRTDepositPool.sol:323]: nit: Should be `LRTOperator`

File: contracts/NodeDelegator.sol [hunk 2]
```diff
@@ -72,18 +74,53 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
         whenNotPaused
         nonReentrant
         onlySupportedAsset(asset)
-        onlyLRTManager
+        onlyLRTOperator
```
Comment (shahthepro) [inline contracts/NodeDelegator.sol:77]: nit: The comment needs to be updated as well

File: contracts/LRTDepositPool.sol (hunk 1 above)
Comment (naddison36) [reply contracts/LRTDepositPool.sol:323]: good spot. I'll fix

File: contracts/NodeDelegator.sol (hunk 2 above)
Comment (naddison36) [reply contracts/NodeDelegator.sol:77]: I'll fix. Thanks

File: contracts/LRTDepositPool.sol [hunk 3]
```diff
@@ -319,6 +319,44 @@ contract LRTDepositPool is ILRTDepositPool, LRTConfigRoleChecker, PausableUpgrad
...
+    function transferAssetsToNodeDelegator(
+        uint256 ndcIndex,
+        address[] calldata assets
+    )
+        external
+        override
+        nonReentrant
+        onlyLRTOperator
+    {
+        address nodeDelegator = nodeDelegatorQueue[ndcIndex];
```
Comment (DanielVF) [inline contracts/LRTDepositPool.sol:335]: I guess this is okay, since solidity now checks that the lookup is inside the array length. But still, maybe a zero address check on the nodeDelegator just for sanity?

File: contracts/NodeDelegator.sol [hunk 4]
```diff
@@ -64,26 +66,61 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
...
+
+            _depositAssetIntoStrategy(assets[i]);
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    function _depositAssetIntoStrategy(address asset) internal {
```
Comment (DanielVF) [inline contracts/NodeDelegator.sol:107]: Let's add a comment to this that the caller is responsible for verifying that the asset is supported.

File: test/integration/ForkTest.t.sol [hunk 5]
```diff
@@ -207,8 +298,8 @@ contract ForkTest is Test {
         vm.stopPrank();
     }
 
-    function transfer_DelegatorNode(address asset, uint256 amountToTransfer) public {
-        vm.prank(manager);
+    function transfer_DelegatorNode(address asset, uint256 amountToTransfer) internal {
+        vm.prank(Addresses.MANAGER_ROLE);
```
Comment (DanielVF) [inline test/integration/ForkTest.t.sol:302]: I kind of like the shorter `manager` syntax better? We can set the local manager from the addresses.

File: test/integration/ForkTest.t.sol [hunk 6]
```diff
@@ -227,16 +318,35 @@ contract ForkTest is Test {
         nodeDelegator1.depositAssetIntoStrategy(asset);
     }
 
-    function unpauseStrategy(address strategyAddress) private {
-        vm.startPrank(Addresses.EIGEN_UNPAUSER);
-
+    /// @dev unpause an EigenLayer Strategy is currently paused
+    function unpauseStrategy(address strategyAddress) internal {
```
Comment (DanielVF) [inline test/integration/ForkTest.t.sol:322]: These unpause things are unrelated to the rest of the PR?

File: contracts/LRTDepositPool.sol (hunk 3 above)
Comment (naddison36) [reply contracts/LRTDepositPool.sol:335]: Good idea. I'll add

File: test/integration/ForkTest.t.sol (hunk 5 above)
Comment (naddison36) [reply test/integration/ForkTest.t.sol:302]: I wanted to be explicit about what was being used. Before the `manager` and `operator` variables were set to the multisig address which will not be the case after this change.

File: test/integration/ForkTest.t.sol (hunk 6 above)
Comment (naddison36) [reply test/integration/ForkTest.t.sol:322]: We needed to unpause EigenLayer to test the deposits. Not a problem now as its open but it will close again.

Comment (DanielVF) [reply test/integration/ForkTest.t.sol:322]: Makes sense!

**Review by DanielVF — DISMISSED** [review]

**Comment by DanielVF** [comment]
@naddison36 Could we add `UtilLib.checkNonZeroAddress(nodeDelegator);` to the multi transfer case as well?

**Review by DanielVF — APPROVED** [review]
